storage/backup/backup-cli/src/coordinators/backup.rs

```diff
@@ -369,46 +420,52 @@ impl BackupCompactor {
         let files = metaview.get_file_handles();
 
         info!("Start compacting backup metadata files.");
-        // Get all compacted chunks to be written back to storage
-        let mut new_files: HashSet<String> = HashSet::new(); // record overwrite file names
+        let mut new_files: HashSet<FileHandle> = HashSet::new(); // record overwrite file names
         for range in metaview.compact_epoch_ending_backups(self.epoch_ending_file_compact_factor)? {
             let (epoch_range, file_name) =
                 Metadata::compact_epoch_ending_backup_range(range.to_vec())?;
-            self.storage
+            let file_handle = self
+                .storage
                 .save_metadata_lines(&file_name, epoch_range.as_slice())
                 .await?;
-            new_files.insert(file_name.to_string());
+            new_files.insert(file_handle);
         }
         for range in metaview.compact_transaction_backups(self.transaction_file_compact_factor)? {
             let (txn_range, file_name) =
                 Metadata::compact_transaction_backup_range(range.to_vec())?;
-            self.storage
+            let file_handle = self
+                .storage
                 .save_metadata_lines(&file_name, txn_range.as_slice())
                 .await?;
-            new_files.insert(file_name.to_string());
+            new_files.insert(file_handle);
         }
         for range in metaview.compact_state_backups(self.state_snapshot_file_compact_factor)? {
             let (state_range, file_name) =
                 Metadata::compact_statesnapshot_backup_range(range.to_vec())?;
-            self.storage
+            let file_handle = self
+                .storage
                 .save_metadata_lines(&file_name, state_range.as_slice())
                 .await?;
-            new_files.insert(file_name.to_string());
+            new_files.insert(file_handle);
         }
 
-        // Move the compacted metadata files to the metadata backup folder
-        for file in files {
+        // Move expired files to the metadata backup folder
+        let (to_move, compaction_meta) =
+            self.update_compaction_timestamps(&mut metaview, files, new_files)?;
+        for file in to_move {
             // directly return if any of the backup task fails
             info!(file = file, "Backup metadata file.");
-            let name = Path::new(&file)
-                .file_name()
-                .and_then(OsStr::to_str)
-                .ok_or_else(|| format_err!("cannot extract filename from {}", file))?;
-            if !new_files.contains(name) {
-                self.storage.backup_metadata_file(&file).await?
-            }
+            self.storage.backup_metadata_file(&file).await?
         }
-
+        // save the metadata compaction timestamps
+        let now = duration_since_epoch().as_secs();
```

Comment:
shouldn't use a new timestamp?

Reply:
per the previous discussion, we want to generate a new compaction_timestamps file instead of overwriting the existing one after each compaction?

Reply:
Inside `self.update_compaction_timestamps` you've already called `duration_since_epoch`, that timestamp is in `compaction_meta`, and that's what you use for the filename (by calling `compact_meta.name()`). And that's why `get_compaction_timestamps_file_name` should be deleted.

Reply:
delete the function and use name() instead